transform-plugins/src/test/java/co/cask/hydrator/plugin/ValueMapperTest.java

```diff
@@ -0,0 +1,715 @@
+
+    Map<String, String> sourceproperties = new ImmutableMap.Builder<String, String>()
+      .put("mapping", "designationid:designation_lookup_table:designationName")
+      .put("defaults", "designationid:DEFAULTID")
+      .put("name", "FlowPlugin").build();
+
+    ETLStage transform = new ETLStage("transform",
+                                      new ETLPlugin("ValueMapper", Transform.PLUGIN_TYPE, sourceproperties, null));
+
+    String sinkTable = "output_table";
+    ETLStage sink = new ETLStage("sink", MockSink.getPlugin(sinkTable));
+
+    ETLBatchConfig etlConfig = ETLBatchConfig.builder("* * * * *")
+      .addStage(source)
+      .addStage(transform)
+      .addStage(sink)
+      .addConnection(source.getName(), transform.getName())
+      .addConnection(transform.getName(), sink.getName())
+      .build();
+
+
+    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(ETLBATCH_ARTIFACT, etlConfig);
+    Id.Application appId = Id.Application.from(Id.Namespace.DEFAULT, "valuemappertest");
+    ApplicationManager appManager = deployApplication(appId, appRequest);
+
+    addDatasetInstance(KeyValueTable.class.getName(), "designation_lookup_table");
+    DataSetManager<KeyValueTable> dataSetManager = getDataset("designation_lookup_table");
+    KeyValueTable keyValueTable = dataSetManager.get();
+    keyValueTable.write("1".getBytes(Charsets.UTF_8), "SE".getBytes(Charsets.UTF_8));
+    keyValueTable.write("2".getBytes(Charsets.UTF_8), "SSE".getBytes(Charsets.UTF_8));
+    keyValueTable.write("3".getBytes(Charsets.UTF_8), "ML".getBytes(Charsets.UTF_8));
+    dataSetManager.flush();
+
+    DataSetManager<Table> inputManager = getDataset(inputTable);
+    List<StructuredRecord> input = ImmutableList.of(
+      StructuredRecord.builder(inputSchema).set(ID, "100").set(NAME, "John").set(SALARY, "1000").set
+        (DESIGNATIONID, "1").build(),
+      StructuredRecord.builder(inputSchema).set(ID, "101").set(NAME, "Kerry").set(SALARY, "1030").set
+        (DESIGNATIONID, "2").build(),
+      StructuredRecord.builder(inputSchema).set(ID, "102").set(NAME, "Mathew").set(SALARY, "1230").set
+        (DESIGNATIONID, "3").build()
+    );
+    MockSource.writeInput(inputManager, input);
+
+    MapReduceManager mrManager = appManager.getMapReduceManager(ETLMapReduce.NAME);
+    mrManager.start();
+    mrManager.waitForFinish(5, TimeUnit.MINUTES);
+
+    DataSetManager<Table> outputManager = getDataset(sinkTable);
+    List<StructuredRecord> outputRecords = MockSink.readOutput(outputManager);
+
+    Map<String, String> nameDesignationMap = new HashMap<String, String>();
+    nameDesignationMap.put("John", "SE");
+    nameDesignationMap.put("Kerry", "SSE");
+    nameDesignationMap.put("Mathew", "ML");
+
+    Assert.assertEquals(3, outputRecords.size());
+    Assert.assertEquals(nameDesignationMap.get(outputRecords.get(0).get(NAME)), outputRecords.get(0).
+      get(DESIGNATIONNAME));
```

Comment:
for future reference, we have the period on the same line as the method. So:

```
Assert.assertEquals(nameDesignationMap.get(outputRecords.get(0).get(NAME)), outputRecords.get(0)
  .get(DESIGNATIONNAME));
```
or
```
Assert.assertEquals(nameDesignationMap.get(outputRecords.get(0).get(NAME)),
                                 outputRecords.get(0).get(DESIGNATIONNAME));
``` 

Though for something like this, you don't need to assert entry by entry.  You can just create the expected map and assert the maps are equal.